### PR TITLE
REXLaunchpad referral integration #103

### LINKED ISSUE #103: REX Launchpad Referral Integration
This task involves integrating the REX Referral system into the REX Launchpad contracts, depends on #102.

### PR DESCRIPTION
**Pull request by gitChimp88**
Started creation of REXLaunchpad.test.ts in order to test upcoming integration of referral in to ricochet launchpad.

Fixes: #103

### CODE REVIEW

**Comment by mikeghen** [comment]
This is looking good, do the tests pass? Is it ready for review?

**Comment by gitChimp88** [comment]
> This is looking good, do the tests pass? Is it ready for review?

@mikeghen Hey! yeah the test passes and it's ready for review.

File: test/REXLaunchpad.test.ts [hunk 1]
```diff
@@ -0,0 +1,440 @@
...
+            expect(
+                (await launchpad.getIDAShares(ETHX_SUBSCRIPTION_INDEX, aliceSigner.address)).toString()
+            ).to.equal(`true,false,980000,0`);
+            // Admin and Carl split 2% of the shares bc of the 50% referral fee
+            expect(
+                (await launchpad.getIDAShares(ETHX_SUBSCRIPTION_INDEX, adminSigner.address)).toString()
+            ).to.equal(`true,false,10000,0`);
+            expect(
+                (await launchpad.getIDAShares(ETHX_SUBSCRIPTION_INDEX, bobSigner.address)).toString()
+            ).to.equal(`true,false,10000,0`);
```
Comment (mikeghen) [inline test/REXLaunchpad.test.ts]: Can you add to this, close the stream and add expects for: 
- confirming the affiliate shares go to 0 on stream close
- open a stream without a referral and confirm it doesn't have an affiliate (DAO gets all the feeshares)

**Review by mikeghen — CHANGES_REQUESTED** [review]

File: test/REXLaunchpad.test.ts (hunk 1 above)
Comment (gitChimp88) [reply test/REXLaunchpad.test.ts]: @mikeghen no problem, changes made!

**Pull request closed without merging**